software/contrib/master_clock.py

```diff
@@ -0,0 +1,340 @@
+        self.activeOption = 1
+
+        # Get working vars
+        self.loadState()
+        self.calcSleepTime()
+        self.getPulseWidth()
+
+        # Get asyncio event loop object
+        #self.el = asyncio.get_event_loop()
+
+        self.tasks = []
+        for n in range(6):
+            self.tasks.append(0)
+
+        # Starts/Stops the master clock
+        @b1.handler_falling
+        def StartStop():
+            self.running = not self.running
+
+        # Cycle screen and toggle config mode
+        @b2.handler_falling
+        def cycleScreen():
+            if ticks_diff(ticks_ms(), b2.last_pressed()) > 500 and ticks_diff(ticks_ms(), b2.last_pressed()) < 4000:
+                self.configMode = not self.configMode
+                # [temp] keep the clock running when in config mode to see if the previous slowdown bug occurs in testing
+                #self.running = False
+                if not self.configMode:
+                    # config mode has just been turned off, save state and lock k2
+                    self.saveState()
+                    self.k2Unlocked = False
+            else:
+                self.k2Unlocked = False
+                
+                # Turn off config mode to avoid current knob positions messing up other settings on the next screen
+                if self.configMode:
+                    self.configMode = False
+                    self.saveState()
+
+                if self.screen == 1:
+                    self.screen = 2
+                elif self.screen == 2:
+                    self.screen = 1
+                else:
+                    self.screen = 1
+
+        # Reset step count upon receiving a din voltage
+        @din.handler
+        def resetDin():
+            self.step = 1
+
+    '''Show running status'''
+    def screen1(self):
+        oled.fill(0)
+        oled.text(str(self.completedCycles) + ':' + str(self.step), 0, 0, 1)
+        if not self.running:
+            oled.text('B1:Start', 0, 23, 1)
+        else:
+            oled.text('B1:Stop', 0, 23, 1)
+        oled.show()
+
+    '''config screen'''
+    def screen2(self):
+        # k1 adjusts selected option
+        self.activeOption = k1.choice([1, 2, 3, 4, 5, 6, 7, 8])
+
+        oled.fill(0)
+        if self.configMode and self.activeOption != 3:
+            configMarker = '|'
+            
+            # if active config option changes, lock k2 and save state
+            if self.previousActiveOption != self.activeOption:
+                self.k2Unlocked = False
+                self.saveState()
+
+            if self.activeOption == 1:
+                # read current knob value
+                newBpm = self.MIN_BPM + k2.read_position(steps=(self.MAX_BPM - self.MIN_BPM + 2))
+                # unlock the knob if it has reached near the same value - avoids messy UX
+                if abs(newBpm - self.state.get('bpm')) <= 10:
```

Comment:
The `LockableKnob` class should cover this functionality for you. Did you get a chance to look at it?